### ISSUE TITLE
NHibernate.StaleStateException: Unexpected row count: 0; expected: 1

### ISSUE THREAD

**Issue by Strubbl** [lformella/xdcc-grabscher#63]
I get a lot of the following exceptions:

```
20-06-2014 12:51:51,842 FATAL [DefaultQuartzScheduler_Worker-6] 0:XG.DB.Dao.WriteToDatabase() changed 
NHibernate.StaleStateException: Unexpected row count: 0; expected: 1
  at NHibernate.AdoNet.Expectations+BasicExpectation.VerifyOutcomeNonBatched (Int32 rowCount, IDbCommand statement) [0x00000] in <filename unknown>:0 
  at NHibernate.AdoNet.NonBatchingBatcher.AddToBatch (IExpectation expectation) [0x00000] in <filename unknown>:0 
  at NHibernate.Persister.Entity.AbstractEntityPersister.Delete (System.Object id, System.Object version, Int32 j, System.Object obj, NHibernate.SqlCommand.SqlCommandInfo sql, ISessionImplementor session, System.Object[] loadedState) [0x00000] in <filename unknown>:0 
```

Every minute I get one to two of those exceptions. Is something wrong with my database? How can I figure out what?

**Comment by lformella** [comment]
The best option is to shutdown XG and start again. Maybe you lose some informations, but the database itself should be ok. Sometimes nhibernate throws these exceptions and I could not finally find out the exact reason. Because of this (and the speed) I switched to db4o in the next XG version.

**Comment by lformella** [comment]
Fixed in [v3.3.0.0](https://github.com/lformella/xdcc-grabscher/releases/tag/v3.3.0.0)
